irida_staramr_results/amr_downloader.py

```diff
@@ -46,13 +61,46 @@ def _write_to_excel(result_files, output_file_name):
     logging.info(f"Creating a new file {output_file_name}.")
     with pd.ExcelWriter(output_file_name, engine='xlsxwriter') as writer:
         # append data frame to file
-        for file in result_files:
-            file_content = file.get_contents()
-            file_sheet_name = file.get_sheet_name()
+        for file_sheet_name in data_frames:
+            logging.debug(f"Writing {file_sheet_name} data to {output_file_name}.")
+            data_frames[file_sheet_name].to_excel(writer, sheet_name=file_sheet_name, index=False)
 
-            logging.info(f"Appending {file_sheet_name} data to {output_file_name}.")
-            data_frame = _convert_to_df(file_content)
-            data_frame.to_excel(writer, sheet_name=file_sheet_name, index=False)
+
+def _files_to_data_frames(results_files):
+    """
+    Accepts a list of results files and returns them as dictionary of filename:dataframe key:value pairs
+    :param results_files:
+    :return:
+    """
+    data_frames = {}
+    for file in results_files:
+        data_frames[file.get_sheet_name()] = _convert_to_df(file.get_contents())
+
+    return data_frames
+
+
+def _append_file_data_to_existing_data_frames(results_files, data_frames):
+    """
+    Accepts a list of results files and appends the data to a given list of data_frames.
+    The data_frames can be an empty dict
+    :param results_files: a list of files to be converted into dataframes, and added or appended to the data_frames dict
+    :param data_frames: a dictionary of filename:dataframe pairs
+    :return: list of dataframe objects with data from input appended, one per unique file name
```

Comment:
This `:return:` description was a bit confusing for me. I understand that it returns a list of dataframes (in a form of a dictionary with sheetname:dataframe pairs) with the new appended data. I'm suggesting changing it to:

"an updated dictionary of dataframe objects containing the newly appended data per file name"..?

Reply:
That makes much more sense than what I had, great suggestion. 
I also added an example of what the dict looks like when it is returned so it's more clear.